src/omp/exec_dist.f90

```diff
@@ -93,22 +93,23 @@ subroutine exec_dist_transeq_compact( &
     real(dp), dimension(:, :), allocatable :: ud, ud_recv_s, ud_recv_e
     real(dp) :: nu
     integer, intent(in) :: nproc, pprev, pnext
-    integer, intent(in) :: n_block
+    integer, intent(in) :: n_groups
+    integer, intent(in) :: sz
 
     integer :: n_data, n_halo
     integer :: k, i, j, n
 
     ! TODO: don't hardcode n_halo
     n_halo = 4
     n = tdsops_d2u%n
-    n_data = SZ*n_block
+    n_data = sz*n_groups
 
-    allocate (ud(SZ, n))
-    allocate (ud_recv_e(SZ, n_halo))
-    allocate (ud_recv_s(SZ, n_halo))
+    allocate (ud(sz, n))
+    allocate (ud_recv_e(sz, n_halo))
```

Comment:
Another minor example, if we have both `sz` and `n_halo` as a parameter, it'll probably be better for performance as the size of the allocation can be determined at the compile time. `n_halo` is not a parameter yet, but we can actually make it, its always a constant value in backends and there are places we hardcoded things based on `n_halo=4`.